Dockerfile

```diff
@@ -1,18 +1,59 @@
-FROM ubuntu:xenial
+# syntax=docker/dockerfile:experimental
 
-RUN apt-get update && apt-get install --yes nginx
+# Build stage: Install ruby dependencies
+# ===
+FROM ruby:2.6-slim AS ruby-dependencies
+RUN apt-get update && apt-get install --no-install-recommends -y build-essential ruby-dev ruby-bundler
+
+# Build stage: Install yarn dependencies
+# ===
+FROM node:10-slim AS yarn-dependencies
+WORKDIR /srv
+ADD package.json .
+RUN --mount=type=cache,target=/usr/local/share/.cache/yarn yarn install
 
-# Set git commit ID
-ARG REVISION_ID
-RUN test -n "${REVISION_ID}"
 
-# Copy over files
+# Build stage: Run "yarn run build-jekyll"
+# ===
+FROM ruby-dependencies AS build-jekyll
```

Comment:
I don't see the sense of keeping the `ruby-dependencies` as its own stage. Any `apt install` layer will probably be cached anyway...